packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -304,6 +359,24 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin
   ScrollPosition get position => _position;
   ScrollPosition _position;
 
+  /// Returns the scroll increment for a single line, for use when scrolling
+  /// using a hardware keyboard.
+  double getIncrement({ ScrollIncrementType type = ScrollIncrementType.line }) {
+    assert(type != null);
+    double defaultScrollIncrement() {
```

Comment:
I find the structure of this method slightly confusing. Maybe it would be clearer to just do:

```dart
if (widget.incrementCalculator != null) {
  return widget.incrementCalculator(...);
}
switch (type) {
  // ...
}
```